### ISSUE TITLE
Additional RDM Responder tests to add

### ISSUE THREAD

**Issue by nomis52** [OpenLightingProject/ola#848]
- [ ] Check the device doesn't respond to different sub-start-codes
- [ ] Supported params for sub devices must return all PIDs
- [ ] Check what happens if you send a DISC command with a non-0 sub device
- [ ] personality 0 / 0 should be an advisory , given eric's reading of the standard
- [ ] check what happens if we find a 'hidden' sensor, prob. print an advisory.

**Comment by peternewman** [comment]
Regarding "check what happens if we find a 'hidden' sensor, prob. print an advisory. " We return "GetSensorDefinition: Found too many 2/1 sensors", so I think we can tick that off.

**Comment by nomis52** [comment]
Indeed.

**Comment by peternewman** [comment]
Although I think in this case the error was not returning enough sensors in DEVICE_INFO, rather than intentionally trying to hide it.

**Comment by peternewman** [comment]
We should also try setting an invalid date.

**Comment by peternewman** [comment]
We should also test factory defaults, or see if they change when we've set other stuff.

**Comment by nomis52** [comment]
Alright, the idea I'm exploring is:

i) separate the tests out into 'standards', i.e. the document the test relates to
ii) parametrize the tests by sub device. We'll need a root & current device context
iii) Allow the user to specify which sub devices to run the test on. Root only, Root & known sub devices, all subdevices.

**Comment by peternewman** [comment]
Can we not do i based on adding the information to the PIDs as discussed previously?

ii and iii sound good. I was going to add some similar stuff for e.g. sensors to run the test against every sensor versus known ones, although that idea probably falls down for the 32 bit stuff like network interface IDs due to time constraints.

If we're reworking, is it also worth adding an option to just run non-intrusive tests, e.g. gets, so you can run it "safely" on something?

**Comment by nomis52** [comment]
I don't want to get into what's "safe" and what isn't.

**Comment by peternewman** [comment]
Fair point, I guess an out of range get could upset something for example.

**Comment by peternewman** [comment]
More test work being done here https://github.com/OpenLightingProject/ola/pull/1067

**Comment by peternewman** [comment]
If a test fails due to timeout, look at retrying after a short delay (flag as warning), try again, then if it still fails move on.

**Comment by peternewman** [comment]
A few more jobs ticked off based on commits in #1067 and #1181 among others.

**Comment by peternewman** [comment]
7.4.1 Clearing of Mute Flag on reset command message (RESET_DEVICE) sent to responder.

**Comment by peternewman** [comment]
Check that the PID we sent is the same one the fixture responds with! Does this work for ACK_TIMER/QUEUED MESSAGE stuff?

**Comment by peternewman** [comment]
Broad and vendorcast DISC_MUTE and DISC_UNMUTE from https://github.com/mathertel/DmxSerial2/issues/10

**Comment by rodb70** [comment]
Verify responder ignores bad command class values.  From Europe 2018 plugfest. 

Make sure the responder only replies to E120_DISCOVERY_COMMAND, E120_GET_COMMAND and E120_SET_COMMAND all other command class ID's should be ignored / timeout.

**Comment by peternewman** [comment]
Clock tests, see https://github.com/OpenLightingProject/ola/issues/1510

**Comment by peternewman** [comment]
> date

See https://github.com/OpenLightingProject/ola/issues/1510

**Comment by peternewman** [comment]
> We should also try setting an invalid date.

See https://github.com/OpenLightingProject/ola/issues/1510

**Comment by peternewman** [comment]
Parameter description check should validate the range of the min/max/default based on the data type.

**Comment by peternewman** [comment]
Check that devices which return queued messages actually support them (assuming this is in the standard), see here: https://groups.google.com/d/msg/open-lighting/Yt_UI57-pyw/U3SeOJ9oCQAJ

**Comment by peternewman** [comment]
Check descriptions are unique (for DMX personalities, sensors etc)

**Comment by peternewman** [comment]
Add at least an advisory (check the standard) if multiple DMX personalities listed in device info, but the relevant PIDs aren't supported.

**Comment by peternewman** [comment]
> ii) parametrize the tests by sub device. We'll need a root & current device context
> iii) Allow the user to specify which sub devices to run the test on. Root only, Root & known sub devices, all subdevices.

Test set of unsupported subdevices too in this model.

**Comment by peternewman** [comment]
Lamp hours/device hours advisories, I suspect lamp shouldn't be more than device, as it says neither should reset on boot.

Maybe try setting lamp hours/device hours/lamp strikes to a non-zero number (both bigger and smaller than the current value)

**Comment by peternewman** [comment]
Advisory if OLA doesn't recognise a non-manufacturer specific PID.
